packages/teleport/src/components/BannerList/Banner.tsx

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React, { ReactNode } from 'react';
+import styled from 'styled-components';
+
+import { Box, Flex, Text } from 'design';
+import { Cross, Info, Warning } from 'design/Icon';
+
+type Props = {
+  message: string;
+  // The backend defines the severity as an integer value with the current
+  // pre-defined values: LOW: 0; MEDIUM: 5; HIGH: 10
+  severity: number;
+  id: string;
+  onClose: (id: string) => void;
+};
+
+export function Banner({ id, message = '', severity = 0, onClose }: Props) {
+  return (
+    <Box bg={getBackgroundColor(severity)} p={3}>
+      <Flex alignItems="center">
+        {generateIcon(severity)}
+        <Text bold>{message}</Text>
+        <Close
+          onClick={() => {
+            onClose(id);
+          }}
+          role="button"
+        >
+          <Cross />
+        </Close>
+      </Flex>
+    </Box>
+  );
+}
+
+function generateIcon(severity: number): ReactNode {
+  // severity is checked using ranges to allow additional severities to be added
+  // on the backend without requiring an update to the client.
```

Comment:
I think we should move backend-specific code out of the reusable UI component.
```ts
// component props
severity: 'info' | 'warning' | 'danger';
```

```ts
// client usage
const mapSeverity = (severity: number) => {
    if (severity < 5) {
      return "info"
    }
    if (severity < 10) {
       return "warning"
    }
    return "danger"
}

const banners = props.alerts
    ? props.alerts.map(alert => ({
        message: alert.spec.message,
        severity: mapSeverity(alert.spec.severity),
        id: alert.metadata.name,
      }))
    : [];

<Banner baners={baners} />
```

Reply:
I went back and forth on this myself but you've convinced me... updated 😁